File: program/program/find-largest-and-smallest-numbers-in-an-array/FindLargestAndSmallestNumbersInAnArray.fs
let FindLargestAndSmallestNumbersInAnArray (input: int list) =
    let sortedInput = List.sort input
    let smallest = sortedInput.[0]
    let largest = sortedInput.[sortedInput.Length-1]
    printfn "Smallest Number in the array: %d" smallest
    printfn "Largest Number in the array: %d" largest

let input = [1; 2; 3; 9; 6; 4; 5]
FindLargestAndSmallestNumbersInAnArray input

File: program/program/find-largest-and-smallest-numbers-in-an-array/find_largest_and_smallest_numbers_in_an_array.go
package main

import "fmt"

func main() {
    var arrsize, i, minPosition, maxPosition int

    fmt.Print("Enter the Array Size to find Smallest & Largest = ")
    fmt.Scan(&arrsize)

    arr := make([]int, arrsize)

    fmt.Print("Enter the Array Items  = ")
    for i = 0; i < arrsize; i++ {
        fmt.Scan(&arrsize[i])
    }
    largest := arr[0]
    smallest := arr[0]

    for i = 0; i < arrsize; i++ {
        if largest < arr[i] {
            largest = arr[i]
            maxPosition = i
        }
        if smallest > arr[i] {
            smallest = arr[i]
            minPosition = i
        }
    }
    fmt.Println("\nThe Largest Number in this Array    = ", largest)
    fmt.Println("The Index Position of Largest Number = ", maxPosition)
    fmt.Println("\nThe Smallest Number in this Array    = ", smallest)
    fmt.Println("The Index Position of Smallest Number = ", minPosition)
}
